fragdenstaat_de/fds_donation/cms_plugins.py

```diff
@@ -191,14 +191,14 @@ def get_percentage(self, amount, max):
         return 0
 
     def get_donated_amount(self, instance):
+        qs = DefaultDonation.objects
+
+        if instance.purpose:
+            qs = qs.filter(purpose=instance.purpose)
         if instance.received_donations_only:
-            qs = DefaultDonation.objects.estimate_received_donations(
-                instance.start_date
-            )
+            qs = qs.estimate_received_donations(instance.start_date)
```

Comment:
I think this is a manager method that is not available on the queryset (ie it cannot be chained). I would put the purpose filter after the if block on the resulting queryset.